### ISSUE TITLE
DatePicker - On reflowing the page, unable to view the entire Calendar details

### ISSUE THREAD

**Issue by MdAmeerMind** [microsoft/fluentui#12220]
Actual behavior:
If page on browser re-flowing upto 400% zoom, then Month and Year selection area are not completely visible.

Expected behavior:
Month and Year selection area should be visible or vertical scroll should be come so that user will move to all value using mouse 

Repro: [https://developer.microsoft.com/en-us/fabric#/controls/web/datepicker](https://developer.microsoft.com/en-us/fabric#/controls/web/datepicker)  

Repro steps:
1) Open above link and zoom up to 400%
2) Now observer Month and Year selection section, it is not completely visible.

**Comment by xugao** [comment]
@lorejoh12 - thoughts on how to handle this ?
also, datepicker looks broken to me on mobile:
![image](https://user-images.githubusercontent.com/1207059/76105728-630ad000-5f8a-11ea-87de-fa671f124790.png)

**Comment by lorejoh12** [comment]
We used to try to have responsive CSS that, if the screen was too small to hold the entire component, it made it into the collapsed state. But it caused a lot of other issues, so we eliminated it. Specifically for mobile view, there's a prop to handle that, `showMonthPickerAsOverlay`, which will allow you to click/tap on the "March 2020" in order to switch between the day picker and the month/year pickers.

For the 400% zoom issue, I'm not really sure what the best solution to this is. Looks like the problem is that the control doesn't responsively shrink beyond 441px width, so when the screen is smaller than that, no scroll bar shows up. We can either set the callout or the DatePicker itself to have a max-width of 100vw to avoid that issue I suppose. I'd like to avoid getting back into having CSS selectors that try to dynamically hide/show content based on screen size like it used to though.

**Comment by MdAmeerMind** [comment]
Hi, 
any update on this issue.

**Comment by MdAmeerMind** [comment]
Hi @lorejoh12 , any update on this issue?

**Comment by lorejoh12** [comment]
Given that the component is prop driven, and there is a prop for having the day and month pickers overlap, I would say that it is the consumer's responsibility to attach a listener to the component for screen size changes/zoom and update the prop as necessary. I'd rather not add that functionality into the control itself because then the control is changing how it interprets the props it's given, rather than just doing what it's told, although I'm open to reviewing a PR for that functionality if someone would like to add it.

The fact that the website isn't mobile friendly with the control means that we might consider adding something to the example page to fix it, but I don't know if we have that requirement yet, Elizabeth was talking about it in #12863 but I don't know if we have the 400% requirement yet.

**Comment by lorejoh12** [comment]
closing as by design for now. If a requirement for reflowing at 400% as default behavior is made, we can reopen and consider pushing that functionality into the component itself. Or, feel free to open a PR for it!
